src/Couchbase.Lite.Shared/Query/DatabaseSource.cs

```diff
@@ -60,7 +60,7 @@ internal override string ColumnName
 
         #region Constructors
 
-        public DatabaseSource(Database database, [NotNull]ThreadSafety threadSafety) : base(database, threadSafety)
+        public DatabaseSource([NotNull]Database database, [NotNull]ThreadSafety threadSafety) : base(database, threadSafety)
```

Comment:
Since this is an internal class, I'd do a quick `Debug.Assert(database != null)` here.